crates/wasm_module/src/parse.rs

```diff
@@ -92,6 +92,43 @@ impl Parse<()> for i32 {
     }
 }
 
+/// Decode a signed 64-bit integer from the provided buffer in LEB-128 format
+/// Return the integer itself and the offset after it ends
+fn decode_i64(bytes: &[u8]) -> Result<(i64, usize), ()> {
```

Comment:
turns out LEB-128 is also used in macho, so it might make sense at some point to lift this out further.